### ISSUE TITLE
Bug in fourier_encode (?) 

### ISSUE THREAD

**Issue by marcdumon** [lucidrains/perceiver-pytorch#51]
Thank you for this great implementation. I'm learning a lot from it!

I think I found a problem in the `fourier_encode` method. 
In this line:
https://github.com/lucidrains/perceiver-pytorch/blob/b33aced4e1b266aeb1383e03ab63f0a9951f9126/perceiver_pytorch/perceiver_pytorch.py#L36

the scales are always the same whatever value of parameter `base`.
Example:
```
max_freq = 10, num_bands=6, base = 2
=> scales = [1.0000, 1.3797, 1.9037, 2.6265, 3.6239, 5.0000]

max_freq = 10, num_bands=6, base = 10
=> scales = [1.0000, 1.3797, 1.9037, 2.6265, 3.6239, 5.0000]
```

**Comment by miccooper9** [comment]
![IMG_20210925_101621](https://user-images.githubusercontent.com/40654905/134754604-8384ae6d-104d-4a65-84c5-425da3fcce23.jpg)

This is a snippet from the perceiver paper, looks like the function is correct. Maybe needs to remove the 'base' parameter since it is irrelevant.

**Comment by marcdumon** [comment]
@miccooper9 Thanks!
I read the paper. Is it correct that when the authors wrote "spaced equally" they mean this on a log-scale and not linear? 
In that case the formula is correct indeed but then the base-parameter is obsolete.

**Comment by lucidrains** [comment]
thanks for catching this! i've put in a fix in 0.7.2

**Comment by marcdumon** [comment]
Problem was fixed in version 0.7.3.

**Comment by marcdumon** [comment]
Problem was fixed in version 0.7.3.
